runtime/ts/storage/in-memory-storage.ts

```diff
@@ -52,25 +54,24 @@ class InMemoryKey extends KeyBase {
 const __storageCache = {};
 
 export class InMemoryStorage {
-  _arcId: string;
   _memoryMap: { [index: string]: InMemoryStorageProvider};
   _typeMap: Map<Type, InMemoryCollection>;
   localIDBase: number;
-  constructor(arcId) {
-      assert(arcId !== undefined, 'Arcs with storage must have ids');
-      this._arcId = arcId;
-      this._memoryMap = {};
-      this._typeMap = new Map();
-      this.localIDBase = 0;
-      // TODO(shans): re-add this assert once we have a runtime object to put it on.
-      // assert(__storageCache[this._arc.id] == undefined, `${this._arc.id} already exists in local storage cache`);
-      __storageCache[this._arcId] = this;
+
+  constructor(private readonly arcId: Id) {
+    assert(arcId !== undefined, 'Arcs with storage must have ids');
+    this._memoryMap = {};
+    this._typeMap = new Map();
+    this.localIDBase = 0;
+    // TODO(shans): re-add this assert once we have a runtime object to put it on.
+    // assert(__storageCache[this._arc.id] == undefined, `${this._arc.id} already exists in local storage cache`);
+    __storageCache[this.arcId.toString()] = this;
```

Comment:
does TS complain without the toString() here? There's a well-known implicit stringification that occurs when using non-string keys in objects.